requirements.txt

```diff
@@ -5,7 +5,7 @@ pydotplus==2.0.2
 requests==2.32.2
 pydantic==2.7.1
 rdkit-pypi==2022.9.5
-mordred==1.2.0
+mordred_community==2.0.5
 scikit-learn==1.5.0
 tqdm==4.66.4
 kennard-stone==2.2.1
```

Comment:
General comment: Add to the beggining of the script a docstring describing the purpose of the module. Then add a docstring for the class, explaining all attributes and methods. Finally, each method should have a docstring with a description of the purpose of the method